quipucords/scanner/network/processing/cpu.py

```diff
@@ -74,3 +74,80 @@ class ProcessCpuBogomips(process.Processor):
     def process(output, dependencies=None):
         """Pass the output back through."""
         return get_line(output['stdout_lines'])
+
+
+class ProcessCpuSocketCount(process.Processor):
+    """Process the cpu socket count."""
+
```

Comment:
Quick question what are the KEY, DEPS, & REQUIRE_DEPS variables used for? I don't see them used in the static method, so I was just curious of how they are used outside of the class.

Reply:
For sure! So the processors defined in this file inherit from the processor class defined in  [process.py](https://github.com/quipucords/quipucords/blob/c212985e8ae57f9702e8b71a7414a8412ec5ce36/quipucords/scanner/network/processing/process.py#L170). You can read the comment within to learn more about them. Basically when you set `DEPS` the processor builds a dictionary to pass to the more specific processors (in this case the socket processor) that has the defined dependencies and their values inside. If the flag for `REQUIRE_DEPS` is set to true, it will not process the key you are trying to process if any of the dependencies are not found, so I had to set it to `false` so that it would still pass the values through but not skip processing if they were not available.